addons/mail/static/src/owl/widgets/discuss.js

```diff
@@ -266,6 +303,40 @@ const DiscussWidget = AbstractAction.extend({
     _onClickMobileNewMessage() {
         this.component.doMobileNewMessage();
     },
+    /**
+     * @private
+     */
+    _onClickModerationAccept() {
+        this.env.store.dispatch('moderationAccept',
+            this.component.storeProps.moderationCheckedMessageLocalIds
+        );
+    },
+    /**
+     * @private
+     */
+    _onClickModerationDiscard() {
+        this.component.state.showModerationDiscardDialog = true;
+    },
+    /**
+     * @private
+     */
+    _onClickModerationReject() {
+        this.component.state.showModerationRejectDialog = true;
+    },
+    /**
+     * @private
+     */
+    _onClickSelectAll() {
+        const moderationUncheckedMessageLocalIds = this.component.storeProps.moderationUncheckedMessageLocalIds;
+        this.env.store.dispatch('moderationMessageCheck', moderationUncheckedMessageLocalIds);
+    },
+    /**
+     * @private
+     */
+    _onClickUnselectAll() {
+        const moderationCheckedMessageLocalIds = this.component.storeProps.moderationCheckedMessageLocalIds;
+        this.env.store.dispatch('moderationMessageUncheck', moderationCheckedMessageLocalIds);
```

Comment:
the purpose here is to (un)select all messages of a thread with given domain.
IMO the action should reflect that, so it should pass a `thread` and `domain` instead of list of messages.

Reply:
done